### ISSUE TITLE
Keys released on the frame they are pressed aren't used as inputs

### ISSUE THREAD

**Issue by Endaris** [panel-attack/panel-game#377]
This issue is for the `sceneRefactor` branch.
As far as I can tell, beta is doing this correctly (although not entirely sure cause the code is a bit wack)

```Lua
function inputManager:keyPressed(key, scancode, isrepeat)
  self.allKeys.isDown[key] = KEY_CHANGE.DETECTED
end
 
function inputManager:keyReleased(key, scancode)
  self.allKeys.isDown[key] = KEY_CHANGE.NONE
  self.allKeys.isPressed[key] = KEY_CHANGE.NONE
  self.allKeys.isUp[key] = KEY_CHANGE.DETECTED
end
```

If a key is released on the same frame it was pressed, `isDown` has no key change logged so when we apply the input in `send_controls`, `isDown` and `isPressed` are nil and the key press is lost.

**Comment by JamesVanBoxtel** [comment]
Sounds like a very important unit test to write. We should also verify action happens on press, not release

**Comment by Endaris** [comment]
Implemented with panel-attack/panel-attack#1040
specifically https://github.com/panel-attack/panel-attack/commit/0fb12ad970caa252c9a42751bae1a05d77c1f14f
